Use status for queue fix and optimization

chuck-dbos opened the pull request:
When starting a workflow, look at any existing record in the workflow status table.
If the workflow is already complete (SUCCESS or ERROR) do nothing (let the caller retrieve the result from the handle)
If the workflow is now ENQUEUED, ensure there is a workflow_queue record for it, but do not add such records if the workflow has moved past that state.

kraftp reviewed (commented):
Great catch and fix, this looks good! (and makes a ton of sense)

Could we also add a new test for a scenario like that in https://github.com/dbos-inc/dbos-transact-ts/issues/706 both here and in Python?

chuck-dbos commented:
> Could we also add a new test for a scenario like that in #706 both here and in Python?

I added a test now that failed on old and passes on new.  A bit of unnatural act there, let me know if you have better ideas on how to do this.  Sorry for the clutter too, I updated the test to use v2 API because it is just much easier to launch() and shutdown().

Python is ongoing, the branch exists, I've tested the same thing manually, and need to automate.

kraftp reviewed (approved):
Great fix!
